src/dbg/historycontext.cpp

```diff
@@ -0,0 +1,96 @@
+#include "historycontext.h"
+#include "disasm_helper.h"
+#include "memory.h"
+#include "console.h"
+#include "watch.h"
+#include "_exports.h"
+
+duint HistoryMaxCount = 4096;
+std::deque<HistoryContext> history;
+
+//This will capture the current instruction
+HistoryContext::HistoryContext()
+{
+    DISASM_INSTR instruction;
+    if(GetFullContextDataEx(hActiveThread, &registers) && MemIsValidReadPtr(registers.cip))
```

Comment:
This will kill performance (both `GetFullContextDataEx` and `MemIsValidReadPtr` are hotspots in performance analysis). I think it should be disabled per default until a new debug engine is in place that will cache this data correctly.